web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/AutoDeploy.tsx

```diff
@@ -93,23 +96,23 @@ export function AutoDeploy({ toggleDeployMethod }: DeployServiceProp) {
       return;
     }
 
+    if (selectedSubnetIds.length === 0) {
+      setHasNoSubnets(true);
+      return;
+    } else {
+      setHasNoSubnets(false);
+    }
```

Comment:
It feels like long term it'd be good to have a way to include this validation within `validator.validate()`. Otherwise we're going to end up with dozens of places doing such validations by hand while they could be using a dedicated mechanism for this instead.